packages/core/src/fhircast/index.ts

```diff
@@ -94,13 +106,13 @@ export const FHIRCAST_EVENT_RESOURCES = {
     patient: { resourceType: 'Patient' },
   },
   'DiagnosticReport-select': {
-    report: { resourceType: 'DiagnosticReport' },
-    select: { resourceType: '*', isArray: true },
+    report: { resourceType: 'DiagnosticReport', reference: true },
+    patient: { resourceType: 'Patient', optional: true, reference: true },
+    select: { resourceType: '*', array: true, reference: true },
   },
   'DiagnosticReport-update': {
-    report: { resourceType: 'DiagnosticReport' },
-    patient: { resourceType: 'Patient', optional: true },
-    study: { resourceType: 'ImagingStudy', optional: true },
+    report: { resourceType: 'DiagnosticReport', reference: true },
```

Comment:
`report` and `patient` are also references for `DiagnosticReport-update`:
https://build.fhir.org/ig/HL7/fhircast-docs/3-6-3-DiagnosticReport-update.html